// SPDX-FileCopyrightText: 2025 Carnegie Mellon University - Living Edge Lab
//
// SPDX-License-Identifier: GPL-2.0-only

syntax = "proto3";
package protocol.result;

import "common.proto";
import "google/protobuf/duration.proto";
import "google/protobuf/timestamp.proto";


/*
 * Compute results generated by datasink modules
 */
message FrameResult {
  string type = 1;
  uint64 frame_id = 2; // For correlation
  repeated ComputeResult result = 3;
}

message ComputeResult {
  oneof type{
    DetectionResult detection_result = 1;
    AvoidanceResult avoidance_result = 2;
    string generic_result = 3;
  }
}

message Detection {
  uint64 detection_id = 1; // Can be multiple objects per frame
  string class_name = 2;
  double score = 3;
  protocol.common.BoundingBox bbox = 4;
  bool hsv_filter_passed = 5;
}

message DetectionResult {
  repeated Detection detections = 1;
}

message AvoidanceResult {
  double actuation_vector = 1; // Actuation vector towards safe area
}
